src/storage/src/source/types.rs

```diff
@@ -412,7 +412,7 @@ impl SourceMetrics {
                 self.source_id,
                 &partition,
                 offset.offset,
-                timestamp as i64,
+                u64::from(timestamp) as i64,
```

Comment:
Naughty `as`.

Reply:
Agreed, but this function doesn't return an error. I'm going to let the owning team decide what to do here.